### ISSUE TITLE
Cohérence entre MDD de SGB (ici) et dans l'énoncé du lab 1

### ISSUE THREAD

**Issue by fuhrmanator** [profcfuhrmanets/log210-systeme-gestion-bordereau-node-express-ts#26]
Est-ce que https://github.com/profcfuhrmanets/log210-systeme-gestion-bordereau-node-express-ts/blob/master/docs/mdd.puml

est la même chose que la partie SGB dans

https://github.com/profcfuhrmanets/log210-enonce-lab1/blob/main/modeles/mdd_sgb_sga.puml

**Comment by Olon11** [comment]
C'est le même système, mais le premier SGB correspond plus à la structure du code.

**Comment by Olon11** [comment]
Dans le MDD du README, `Grade` est devenu `Mark`. La documentation et le code utilise toujours `Grade`. Est-ce qu'on devrait refactoriser pour utiliser `Mark`?

**Comment by fuhrmanator** [comment]
> Est-ce qu'on devrait refactoriser pour utiliser `Mark`

`Grade` est correct. J'ai changé à `Mark` p-e parce qu'on utilise le terme plus au Canada en anglais, mais c'est effectivement un synonyme. C'est plus simple de remettre "Grade" dans le MDD.
 
Cependant, toute l'histoire de MDD à moitié en anglais (les verbes sont en français) crée des incohérences. Historiquement, c'est à cause de ce projet open-source qu'on a fait en anglais et qui était l'inspiration du serveur SGB (et le projet SGA). https://github.com/fuhrmanator/course-activity-planner/blob/master/ooad/overview.md#domain-model

Le but initial de SGB était de faire un serveur (boîte noire) avec une API comme un "système externe" que l'on ne peut modifié (malgré l'accès au code source...). C'est une hypothèse pour certains patrons comme Proxy ([R1](https://github.com/profcfuhrmanets/log210-enonce-lab1/blob/main/README-exigences-client.md#r1-tol%C3%A9rer-panne-temporaire-de-sgb)) et Decorator. 

Suivre une méthodologie de développement avec un MDD, etc. va à l'encontre de l'idée d'une "boîte noire" et beaucoup de billets dans ce dépôt sont actuellement dus à ça. 

Si on va fournir un MDD de SGB ou utiliser SGB comme exemple de la méthdologie du cours, alors il vaudrait mieux tout faire en français. C'est-à-dire, le MDD ici doit être cohérent avec ce qu'on donne dans SGA, y compris dans la même langue (française).

**Comment by Olon11** [comment]
Effectivement, j'avais remarqué l'incohérence dans l'utilisation de la langue par rapport au jeu de dés. Je crois qu'on devrait utiliser SGB comme exemple de bonne conception pour les étudiants. Donc, il faudra :

- [ ] Traduire le MDD du SGB en français
- [ ] Mettre à jour le MDD du SGB dans l'énoncé du laboratoire
- [ ] Restructurer le code pour qu'il soit en français (correspondre au SGB)

**Comment by Olon11** [comment]
J'ai [traduit le MDD en français](https://www.plantuml.com/plantuml/svg/VL9BRXH13DttAQnU8cMHxIYA8W4AArW0ksngkfo9eThLibsZH8W3S1Fc77qnkdgc8akJOUdFU_NyNjsf1R4ofav_4az1mWXgBmb1JMXO9qyK4N2Sx0L67BSesdL32ghm5LKfC_pg03YFcnN4kwMAys5Rp3KUYL-htmt_tIYH1iCYAmT5XvXaFdXI_uJZwEybbmdVvoBQGr0urj3cdx844Zpn7uUEyyU2S-KJ5l7ZC1ZjoUQ3wuqv-fMKXv1SJV4ZAzAE0zkvt5Nd-R569dY-FV-n4Ycm0JuRSlIMILjgp_r_wtEsPa9ZTVauQrphsqjj41lkKAhAMliyR2cvr3_XGcNQesWrhUkEv_RNFLnTGV-kR-biuCDy68GCRhkMK1tNhMktzUNn4SL0CMM6cxTMtVMRLY34m0g5OLU9O5XLD9BsQvsg4-_wdYner2CIFNIHMariZwzuPoKkKWShA_g1mIJiGtc6QI60l3La4hXf3EUNHXm3HuGKGEP3MEEZp0-rTu7BdUvkMItV0Ag1e4wPDMnJLGMtqBtgVJEJYY7MKoukIB_jVpUEKyg4bvQTFzvQHTdJW7ePTefe0_VpNwdqTyYnZEaV). Je suggère de mettre à jour le dépôt en même temps que #9.
